### PR TITLE
Feature/pypi

### PR DESCRIPTION
**Pull request by pultarmi**
Basic pypi package

Calling now works as "DOCKER_RUN ..." or as "protoc-polyglot ..." python package  call. Python package can be created and uploaded to PyPi repository via "source upload-to-pypi.sh".

Installation of hatchling can be tricky, depending on your OS. I will a readme for that later.

Package is installed via "pip3 install protoc-polyglot". Plugin managing is not yet implemented, hence trying to compile a python package results in this:
```
(uni) ubuntu@ebook:~/work/QS/protoc-polyglot$ protoc-polyglot python protoc bookclub
Plugin missing, downloading...
Failed to download https://github.com/quantasoftcz/protoc-polyglot/files/grpc_python_plugin. Status code: 404
An error occurred: Could not find  release
```

### CODE REVIEW

File: upload-to-pypi.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+# make sure to increase the "post" number: e.g. from 0.0.1-1 to 0.0.1-2
+
+rm -rf ./dist
+hatchling build
+twine upload -p $PYPI_KEY dist/*
+
+#ppackage/protobuf/bin/protoc -I ./samples --plugin=protoc-gen-grpc=ppackage/grpc/grpc_python_plugin --grpc_out=./output/python/bookclub --python_out=./output/python/bookclub bookclub/book.proto bookclub/member.proto bookclub/bookclub.proto
```
Comment (kunah) [inline upload-to-pypi.sh:8]: Delete unnecessary comments
```suggestion
```

File: upload-to-pypi.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+pip install protoc-polyglot --upgrade
+pip install protoc-polyglot --upgrade
```
Comment (kunah) [inline upload-to-pypi.sh:11]: 2 installations?
```suggestion
```

File: protoc_polyglot/cli.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        
+        with os.scandir(self.settings.CORE_DIR) as entries:
+            print('Supported languages:')
+            folders = [entry.name for entry in entries if entry.is_dir() and entry.name not in ['__pycache__']]
+            for f in folders:
+                print(f)
+        
+        # if self.settings.CORE_DIR == '':
+        #     print('Hello world from polyglot')
+        #     exit()
```
Comment (kunah) [inline protoc_polyglot/cli.py:115]: Shouldn't it have a plugin name as an  argument?

**Review by kunah — CHANGES_REQUESTED** [review]
Only code style changes are requested + link a YouTrack Issue

File: protoc_polyglot/cli.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    def get_grpc_path(self) -> str:
+        return "plugins/grpc"
+    
+    def download_grpc_and_protobuf(self):
+        name = f"15369526/grpc-{self.settings.grpc_version}_protobuf-{self.settings.protobuf_version}.zip"
+        if not os.path.exists(self.get_grpc_path()):
+            print("gRPC missing, downloading...")
+        return Tools.download_file(name, self.get_grpc_path())
+        
+    def download_plugin(self):
```
[the remark this answers is not in the document]
Comment (pultarmi) [reply protoc_polyglot/cli.py:98]: Plugin name se vezme z self.get_plugin_path() - to je funkce, ktera cte nazev pluginu ze specializovaneho UI pro jazyk (treba python/cli.py)

File: protoc_polyglot/cli.py (hunk 3 above)
Comment (pultarmi) [reply protoc_polyglot/cli.py:115]: Nevim presne co myslis. Jako tahle funkce je trochu divna, to jo. Nevedel jsem totiz poradne co s ni, protoze napr. Python package se k ni vubec nedostane (podporuje jen protoc-polyglot [language] [command] ... - coz je na dalsi diskuzi)

File: upload-to-pypi.sh (hunk 2 above)
Comment (pultarmi) [reply upload-to-pypi.sh:11]: yep, jedna nestaci, neptej se me proc. Chova se to tak, ze prvni neudela nic, ale zjisti, ze je na remote nova verze, druha ji pak nainstaluje

File: upload-to-pypi.sh (hunk 1 above)
Comment (pultarmi) [reply upload-to-pypi.sh:8]: Pokud to nicemu nevadi, jestli bych si tam comment nechal, protoze se mi hodi jako reference na testovani, ze se vytvari prikaz spravne

File: protoc_polyglot/cli.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            for f in folders:
+                print(f)
+        
+        # if self.settings.CORE_DIR == '':
+        #     print('Hello world from polyglot')
+        #     exit()
+        
+        exit()
+        
+        with open(self.settings.services_yaml, 'r') as file:
```
[the remark this answers is not in the document]
Comment (pultarmi) [reply protoc_polyglot/cli.py:119]: Viz odpoved nahore. Tahle funkce je celkove na diskuzi

File: protoc_polyglot/cli.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pultarmi) [reply protoc_polyglot/cli.py:98]: yes, should be fixed now

File: protoc_polyglot/cli.py (hunk 3 above)
Comment (pultarmi) [reply protoc_polyglot/cli.py]: Ok, it's deleted

**Review by kunah — APPROVED** [review]

**Pull request merged**